### PR TITLE
Fix: ignore trailing whitespace in template literal (fixes #5786)

### LINKED ISSUE #5786: No trailing whitespace should ignore multiline template strings
**What version of ESLint are you using?**

2.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```
{
  "no-trailing-spaces": 2
}
```

**What did you do? Please include the actual source code causing the issue.**

```javascript
let str = `
${a}
      <-- significant trailing whitespace
${b}
`
```

**What did you expect to happen?**

Whitespace could be significant inside multiline template strings, so the rule should ignore it.

**What actually happened? Please include the actual, raw output from ESLint.**

```
t.js:3:1: Trailing spaces not allowed. (no-trailing-spaces)
```

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

File: lib/rules/no-trailing-spaces.js [hunk 1]
```diff
@@ -63,7 +64,10 @@ module.exports = function(context) {
                 linebreaks = context.getSource().match(/\r\n|\r|\n|\u2028|\u2029/g),
```
Comment (mysticatea) [inline lib/rules/no-trailing-spaces.js]: Should we replace deprecated APIs with new APIs in this rule?

- `context.getSource()` → `sourceCode.text`
- `src.split(/\r?\n/)` → `sourceCode.lines`

Comment (platinumazure) [inline lib/rules/no-trailing-spaces.js]: I'm :+1: to replacing deprecated APIs (but don't mind me)

Comment (kaicataldo) [inline lib/rules/no-trailing-spaces.js]: Makes sense to me :)

**Comment by kaicataldo** [comment]
@mysticatea Updated!

**Comment by mysticatea** [comment]
@kaicataldo Thank you! LGTM.

**Comment by gyandeeps** [comment]
@kaicataldo This rule is `fixable` so your invalid test are missing the fixed output?

**Comment by kaicataldo** [comment]
Ah, thanks @gyandeeps. Haven't worked with a fixable rule before, missed that. Will update and ping again. Thanks.

**Comment by kaicataldo** [comment]
Something weird going on with my change and the autofixing (It's removing the last two characters from the template literal : `` }` ``). Will try to get back to this over the weekend.

**Comment by mysticatea** [comment]
Oops, good catch @gyandeeps, thank you.

...So, I'm still afraid clicking the merge button...

**Comment by kaicataldo** [comment]
Ah, figured out what was going on and found and fixed another bug that was occurring. The total length wasn't being updated in my check before continuing to the next iteration of the loop. This also wasn't happening for the `skipBlankLines` check.

The problem with this is that the total length is wrong for the next iteration of the loop, the `fixRange` is actually off. We actually need to account for those whitespace characters in the skipped blank lines for the total length and therefore the autofix range to be correct.

**Comment by kaicataldo** [comment]
Updated with the fix mentioned above (and the correct output from the autofix - thanks for catching that, @gyandeeps!)

**Comment by gyandeeps** [comment]
LGTM :+1: 
Thanks for contributing.

**Pull request merged**